sdk/core/System.ServiceModel.Rest/src/Temporary/OptionalDictionary.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+#nullable enable
+
+namespace System.ServiceModel.Rest.Shared.Core.Serialization
+{
+    public class OptionalDictionary<TKey, TValue> : IDictionary<TKey, TValue>, IReadOnlyDictionary<TKey, TValue> where TKey: notnull
```

Comment:
This and the other Optional* types are fundamental to HLC model serialization today.  I'm curious whether these will be needed when/if we rework the model serialization story, but don't know the outcome yet - we need to preserve this functionality somehow.